### PR TITLE
test: refactor UnicodeJspsTest and  create unicodetools-testutils

### PR DESCRIPTION
**Pull request by srl295**
- create module unicodetools-testutils for shared test utils
- move UnicodeJspsTest into UnicodeJsps/src/test/java

For: #146

### CODE REVIEW

**Comment by srl295** [comment]
@macchiati  and anyone:  test failures start here: https://github.com/unicode-org/unicodetools/runs/5503791433?check_suite_focus=true#step:10:904

**Comment by srl295** [comment]
tests should pass… more fixes can be done in the future.

File: UnicodeJsps/src/test/java/org/unicode/jsptest/TestLanguageid.java [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
         String results;
         results = LanguageCode.validate("pap-CW",new ULocale("en"));
         if (!assertTrue("", results.contains("Curaçao"))) {
             errln(results);
         }
-        
+
         results = LanguageCode.validate("$, eng-840, fr-fr",new ULocale("en"));
-        assertTrue("", results.contains("target='languageid'>fr-FR</b>"));
+        assertContains(results, "target='languageid'>fr-FR</a></b>");
```
Comment (srl295) [inline UnicodeJsps/src/test/java/org/unicode/jsptest/TestLanguageid.java:20]: @macchiati  this seemed like a simple data fix in the assertion, maybe the HTML just changed?

File: UnicodeJsps/src/test/java/org/unicode/jsptest/TestEmoji.java [hunk 2]
```diff
@@ -2,18 +2,20 @@
 
 import java.io.IOException;
 
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledIf;
```
Comment (markusicu) [inline UnicodeJsps/src/test/java/org/unicode/jsptest/TestEmoji.java:6]: unused?

File: UnicodeJsps/src/test/java/org/unicode/jsptest/TestBuilder.java [hunk 3]
```diff
@@ -20,17 +20,20 @@
 
 import com.ibm.icu.dev.test.TestFmwk;
 
+/**
+ * "Not really a test, move"
+ */
```
Comment (markusicu) [inline UnicodeJsps/src/test/java/org/unicode/jsptest/TestBuilder.java:25]: These comments look temporary. Remove before merging?

File: unicodetools-testutils/src/main/java/org/unicode/unittest/TestFmwkMinusMinus.java [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+    public static String toPattern(String title, UnicodeSet primary) {
+        return primary == null ? title : primary.toPattern(false);
+    }
+
+    /**
+     * Copied from TestFmwk. Low level assertion.
+     */
+    public boolean handleAssert(boolean result, String message,
+    Object expected, Object actual, String relation, boolean flip) {
```
Comment (markusicu) [inline unicodetools-testutils/src/main/java/org/unicode/unittest/TestFmwkMinusMinus.java:177]:
```suggestion
    public boolean handleAssert(boolean result, String message,
            Object expected, Object actual, String relation, boolean flip) {
```

**Review by markusicu — COMMENTED** [review]
changes look plausible so far... thanks for doing all of this work!

File: UnicodeJsps/src/test/java/org/unicode/jsptest/TestEmoji.java (hunk 2 above)
Comment (srl295) [reply UnicodeJsps/src/test/java/org/unicode/jsptest/TestEmoji.java:6]: They are used as @ annotations.

File: UnicodeJsps/src/test/java/org/unicode/jsptest/TestBuilder.java (hunk 3 above)
Comment (srl295) [reply UnicodeJsps/src/test/java/org/unicode/jsptest/TestBuilder.java:25]: The comments were on TestAll about these classes. They belong on the classes I think.

**Comment by srl295** [comment]
> changes look plausible so far... thanks for doing all of this work!

Thanks.  Fixed indentation, ok to merge ?

File: UnicodeJsps/src/test/java/org/unicode/jsptest/TestEmoji.java (hunk 2 above)
Comment (markusicu) [reply UnicodeJsps/src/test/java/org/unicode/jsptest/TestEmoji.java:6]: I know, and I see you adding `@Test` and `@EnabledIf` but I don't see `@DisabledIf` so this one looks unused.

**Comment by srl295** [comment]
Ah yes. DisabledIf is indeed unused. Will fix.

**Review by markusicu — APPROVED** [review]
lgtm tnx

**Pull request merged**